tests/test_modeling_common.py

```diff
@@ -162,6 +167,39 @@ def _mock_all_init_weights(self):
         self.tie_weights()
 
 
+@require_read_token
+def _test_torch_compile(ckpt, directory):
+    error = None
+    try:
+        from transformers import AutoTokenizer
+
+        os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+        batch_size = 1
+        n_iter = 3
+
+        tokenizer = AutoTokenizer.from_pretrained(ckpt)
+        model = AutoModelForCausalLM.from_pretrained(ckpt, torch_dtype=torch.float16).to(torch_device)
+
+        model.generation_config.max_new_tokens = 4
+        model.generation_config.max_new_tokens = 4
+
+        model.generation_config.cache_implementation = "static"
+        model.forward = torch.compile(model.forward, mode="reduce-overhead", fullgraph=True)
+
+        input_text = "Why dogs are cute?"
+        input_ids = tokenizer([input_text] * batch_size, return_tensors="pt").to(torch_device)
+
+        for i in range(n_iter):
+            _ = model.generate(**input_ids, do_sample=False)
+
+    except Exception:
+        error = f"{traceback.format_exc()}"
+
+    with open(os.path.join(directory, "results.json"), "w") as fp:
+        json.dump({"error": error}, fp, indent=4)
```

Comment:
is it more readable in json format?

Reply:
no, it's not very important. It's just for the subprocess to report something back to the parent process (in a easy way)

Reply:
But having a dictionary as outputs is good (so use json to save)